Update VPP Device images for ARM

jlinkes opened the pull request:
Added support for aarch64 VPP Device and made the current csit-shim and csit-sut Dockerfiles platform (x86_64 vs aarch64) agnostic.

jlinkes committed:
Update VPP Device containers for ARM

Update csit-shim-arm and add csit-sut-arm.

Signed-off-by: juraj.linkes <juraj.linkes@pantheon.tech>

jlinkes committed:
Merge remote-tracking branch 'upstream/master'

jlinkes committed:
ARM VPP Device updates and cleanup

* Merge csit-shim and csit-shim-arm and slim it down
* Merge csit-sut and csit-sut-arm
* Remove obsolete csit-vpp-device-test

Signed-off-by: juraj.linkes <juraj.linkes@pantheon.tech>

pmikus commented on csit-shim/Dockerfile:
```diff
@@ -1,30 +1,35 @@
-# Configure locales
-RUN locale-gen en_US 
+# Docker install
+RUN curl -fsSL https://get.docker.com | sh
```
Please make it a one command with above install. It will save one layer.

pmikus commented on csit-shim/Dockerfile:
```diff
@@ -1,30 +1,35 @@
-RUN locale-gen en_US 
+# Docker install
+RUN curl -fsSL https://get.docker.com | sh
 
```
Please make it one command with line 15. It will save another layer.

pmikus commented on csit-sut/Dockerfile:
```diff
@@ -23,27 +23,24 @@ ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
+ && dpkg-reconfigure locales
+
+# Base layer
+RUN apt-get install -y -qq \
```
Merge apt-get layers into single RUN

pmikus commented on csit-sut/Dockerfile:
```diff
@@ -116,13 +114,22 @@ RUN pip install \
+RUN chmod 755 /root/get-linux-vm-image.sh
+RUN /root/get-linux-vm-image.sh
+
+RUN rm -rf /var/lib/apt/lists/*
```
Must be part of /root/get-linux-vm-image.sh script + should be also on line 40 of single apt-get

pmikus replied:
The problem is that this command has no effect. Every "RUN" is creating new Docker layer, if you do not make it clean in that layer, you lost access to clean it and resulting file will be huge as each apt-get is downloading lots of data.

jlinkes committed:
Remove unnecessary layers from csit ubuntu images

Signed-off-by: juraj.linkes <juraj.linkes@pantheon.tech>

pmikus commented:
Just for making records. Do you have in plan or did you test those images? Sandbox?

jlinkes commented:
> 
> 
> Just for making records. Do you have in plan or did you test those images? Sandbox?

I tested the arm images. I can test the x86_64 images if I have an environment for it.

pmikus commented on csit-sut/Dockerfile:
```diff
@@ -116,12 +109,14 @@ RUN pip install \
+# ARM workaround
+RUN pip install scipy==1.1.0
+
+# Kernel image for VM tests download and extract
```
Remove

jlinkes replied:
Right, we don't need the kernel image in csit-sut when it's on the host and accessible via mount point. I'll remove it.

pmikus commented on csit-sut/Dockerfile:
```diff
@@ -11,10 +12,9 @@
 
-FROM ubuntu:bionic-20180821
-
+FROM ubuntu:18.04
```
Is this floating build or fixed? IoW will it change in time? Is so i prefer fixed like before.

pmikus commented on csit-sut/get-linux-vm-image.sh:
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
```
remove?

jlinkes replied:
I'll remove the whole script since it's not needed.

jlinkes committed:
Remove download of linux vm image

No longer needed since the containers will have access to hosts' linux
vm images.

Signed-off-by: juraj.linkes <juraj.linkes@pantheon.tech>

jlinkes committed:
Merge remote-tracking branch 'upstream/master'

jlinkes commented:
Csit-sut has been tested in this patch:
https://gerrit.fd.io/r/#/c/20688/

And the actual run:
https://jenkins.fd.io/job/csit-vpp-device-master-ubuntu1804-1n-skx/2553/

pmikus commented:
+1

pmikus reviewed (approved):
- Master image tested and works
- Jenkins executor not touched (ubuntu18)
- Old images cleaned

snergfdio commented on csit-sut/Dockerfile:
```diff
@@ -1,4 +1,5 @@
 # Copyright (c) 2019 Cisco and/or its affiliates.
+# Copyright (c) 2019 PANTHEON.tech s.r.o. and/or its affiliates.
```
can't eat it with this....flip it back to cisco and i'll merge it straight away

jlinkes replied:
Removed.

After you merge it, could you build and push the images? And also setup csit-shim on the vpp device server?

jlinkes committed:
Fix Copyright in csit-sut

Signed-off-by: juraj.linkes <juraj.linkes@pantheon.tech>

jlinkes commented:
Ed, is this ready for merging? If so, please build and push the images and install csit-shim on the tx2 (.69) node.
